### PR TITLE
Created function for perplexity search

### LINKED ISSUE #2: Backend - FastApi Search Page
Create CORS for calling Llama / llama index for converstations with client

- [ ] Create api to check if prompt is detailed enouth (f->b string, b->f bool)

### PR DESCRIPTION
**Pull request by PiotrTyrakowski**
created function for perplexity search in llm_service.py file

### CODE REVIEW

File: backend/server/app/services/product_service.py [hunk 1]
```diff
@@ -20,4 +20,29 @@ async def process_query(self, query: str) -> Dict:
 
     async def update_query(self, query: str, answers: List[str]) -> str:
         updated_query = await self.llm_service.update_issue_details([query] + answers)
-        return updated_query
\ No newline at end of file
+        return updated_query
+    
+     async def find_product(self, consumer_need: str) -> Product:
+        # First, check if the consumer need is detailed enough
+        is_detailed = await self.llm_service.is_prompt_detailed_enough(consumer_need)
```
Comment (Jlisowskyy) [inline backend/server/app/services/product_service.py]: I think we should not perform any 'validation' at this development stage

File: backend/server/app/services/product_service.py [hunk 2]
```diff
@@ -20,4 +20,29 @@ async def process_query(self, query: str) -> Dict:
...
+        
+        if not is_detailed:
+            # If not detailed, get more details
+            questions = await self.llm_service.get_details_questions(consumer_need)
```
Comment (Jlisowskyy) [inline backend/server/app/services/product_service.py]: I don't see where questions are used

File: backend/server/app/websockets/connection_manager.py
Comment (Jlisowskyy) [inline backend/server/app/websockets/connection_manager.py:1]: After investigation probably there is no need for ConnectionManager

File: backend/server/app/websockets/product_finder.py [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
+from .connection_manager import ConnectionManager
 
 router = APIRouter()
-
-# Store active connections
-active_connections = {}
+manager = ConnectionManager()
 
 @router.websocket("/ws/product-finder")
 async def product_finder_websocket(websocket: WebSocket):
```
Comment (Jlisowskyy) [inline backend/server/app/websockets/product_finder.py:10]: This function should be simplified

**Review by Jlisowskyy — DISMISSED** [review]
Small changes are required

File: backend/server/app/services/product_service.py (hunk 1 above)
Comment (Jlisowskyy) [reply backend/server/app/services/product_service.py]: Soluation: temporarly replaced those lines with TODO

File: backend/server/app/services/product_service.py (hunk 2 above)
Comment (Jlisowskyy) [reply backend/server/app/services/product_service.py]: Soluation: temporarly replaced those lines with TODO

File: backend/server/app/websockets/connection_manager.py
Comment (Jlisowskyy) [reply backend/server/app/websockets/connection_manager.py:1]: Solution: replaced with 'event' in the web socket function

File: backend/server/app/websockets/product_finder.py (hunk 3 above)
Comment (Jlisowskyy) [reply backend/server/app/websockets/product_finder.py:10]: Solution: simplification was pushed to the branch

File: backend/server/app/services/product_service.py (hunk 1 above)
Comment (PiotrTyrakowski) [reply backend/server/app/services/product_service.py]: Yes you are right.
This function is expired and we will not use it.

**Review by PiotrTyrakowski — COMMENTED** [review]
everything looks good

**Review by kryczkal — APPROVED** [review]

**Pull request merged**